suite-native/device-authentication/src/deviceAuthenticationSlice.ts

```diff
@@ -0,0 +1,44 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+import { UI } from '@trezor/connect';
+
+import { isPinButtonRequestCode } from './utils';
+
+type DeviceAuthenticationState = {
+    hasDeviceRequestedPin: boolean;
+};
+
+type DeviceAuthenticationRootState = {
+    deviceAuthentication: DeviceAuthenticationState;
+};
+
+const deviceAuthenticationInitialState: DeviceAuthenticationState = {
+    hasDeviceRequestedPin: false,
+};
+
+export const deviceAuthenticationSlice = createSlice({
+    name: 'deviceAuthentication',
+    initialState: deviceAuthenticationInitialState,
+    reducers: {},
+    extraReducers: builder => {
+        builder
+            .addCase(UI.REQUEST_PIN, state => {
+                state.hasDeviceRequestedPin = true;
+            })
+            .addCase(UI.REQUEST_BUTTON, (state, action) => {
+                if (isPinButtonRequestCode(action)) {
```

Comment:
```suggestion
state.hasDeviceRequestedPin = isPinButtonRequestCode(action);
```

Reply:
I think this would make it more readable. Otherwise great job :)

Reply:
In a follow up PR, there will be more stuff that will be related to passphrase. I've played with this but want to keep this simple as it can become really confusing otherwise. I'll revisit your suggestion when I plug passphrase into this to see whether it can be simplified.